e2e_samples/parking_sensors/scripts/deploy_azdo_service_connections_azure.sh

```diff
@@ -30,52 +30,121 @@ set -o errexit
 set -o pipefail
 set -o nounset
 
+
 ###################
 # REQUIRED ENV VARIABLES:
 #
 # PROJECT
 # ENV_NAME
 # RESOURCE_GROUP_NAME
 # DEPLOYMENT_ID
+###############
 
 . ./scripts/common.sh
 
-###############
-# Setup Azure service connection
+###########################################
+# Setup Azure service connection variables
+###########################################
 az_service_connection_name="${PROJECT}-serviceconnection-$ENV_NAME"
-
 az_sub=$(az account show --output json)
 az_sub_id=$(echo "$az_sub" | jq -r '.id')
 az_sub_name=$(echo "$az_sub" | jq -r '.name')
+role="Contributor"
 
-# Create Service Account
-az_sp_name=${PROJECT}-${ENV_NAME}-${DEPLOYMENT_ID}-sp
-log "Creating service principal: $az_sp_name for azure service connection"
-az_sp=$(az ad sp create-for-rbac \
-    --role contributor \
-    --scopes "/subscriptions/${az_sub_id}/resourceGroups/${RESOURCE_GROUP_NAME}" \
-    --name "$az_sp_name" \
-    --output json)
-service_principal_id=$(echo "$az_sp" | jq -r '.appId')
-az_sp_tenant_id=$(echo "$az_sp" | jq -r '.tenant')
-
-# Create Azure Service connection in Azure DevOps
-azure_devops_ext_azure_rm_service_principal_key=$(echo "$az_sp" | jq -r '.password')
-export AZURE_DEVOPS_EXT_AZURE_RM_SERVICE_PRINCIPAL_KEY=$azure_devops_ext_azure_rm_service_principal_key
-
-if sc_id=$(az devops service-endpoint list -o json | jq -r -e --arg sc_name "$az_service_connection_name" '.[] | select(.name==$sc_name) | .id'); then
+
+#Project ID
+project_id=$(az devops project show --project "$AZDO_PROJECT" --organization "$AZDO_ORGANIZATION_URL" --query id -o tsv)
+log "Project ID: $project_id"
+
+# Check if the service connection already exists and delete it if found
+sc_id=$(az devops service-endpoint list --project "$AZDO_PROJECT" --organization "$AZDO_ORGANIZATION_URL" --query "[?name=='$az_service_connection_name'].id" -o tsv)
+if [ -n "$sc_id" ]; then
     log "Service connection: $az_service_connection_name already exists. Deleting service connection id $sc_id ..." "info"
-    az devops service-endpoint delete --id "$sc_id" -y
+    # Get Obj ID 
+    spnAppObjId=$(az devops service-endpoint show --id "$sc_id" --org "$AZDO_ORGANIZATION_URL" -p "$AZDO_PROJECT" --query "data.appObjectId" -o tsv)
+    log "Service Principal App Object ID: $spnAppObjId"
+
+    # Get list of federated credentials
+    spnCredlist=$(az ad app federated-credential list --id "$spnAppObjId" --query "[].id" -o json)
+    log "Federated credentials: $spnCredlist"
```

Comment:
```suggestion
    log "Found existing federated credentials. Deleting..."
```

I would avoid logging that variable as it's an array of strings.

Reply:
done